### ISSUE TITLE
tests should fail for additionalProperties in nested objects

### ISSUE THREAD

**Issue by ScatteredInk** [openownership/data-standard#201]
At the moment, this entityStatement will fail if used in a test:

```
 {
    "extraField": "hello",
    "publicationDetails": {
      "publicationDate": "2019-06-13",
      "publisher": {
        "name": "Example Register"
      },
      "bodsVersion": "0.2"
    },
    "statementID": "60bb3996-b434-4377-930b-39972c65b400",
    "statementType": "entityStatement",
    "statementDate": "2019-05-05",
    "entityType": "registeredEntity",
    "name": "Listed Company OS-17"
}
```

But this entityStatement will pass:

```
 {
    "publicationDetails": {
      "extraField": "hello",
      "publicationDate": "2019-06-13",
      "publisher": {
        "name": "Example Register"
      },
      "bodsVersion": "0.2"
    },
    "statementID": "60bb3996-b434-4377-930b-39972c65b400",
    "statementType": "entityStatement",
    "statementDate": "2019-05-05",
    "entityType": "registeredEntity",
    "name": "Listed Company OS-17"
}
```

This line is adding additionalProperties false only at the root of the object, whereas we want it everywhere. https://github.com/openownership/data-standard/blob/3c113b21fb1cf2fc0d1af2a2378d8ae3a9a083d2/tests/bods_validate.py#L69

This file was built before we had compile-to-jsonschema so maybe worth switching to that here?

**Comment by kd-ods** [comment]
Yes - just saw this: https://github.com/OpenDataServices/compile-to-json-schema/pull/4

**Comment by ScatteredInk** [comment]
Fixed with #251
